src/supervisor/message_handlers.py

```diff
@@ -130,9 +130,10 @@ def _build_message(self, message_details):
 
         #: gzip all attachments and add to message
         for original_path in message_details.attachments:
-            path = Path(original_path)  #: convert to a Path if it isn't already
-            if path.is_file():
-                message.attach(self._build_gzip_attachment(path))
+            if original_path:  #: Path() doesn't like None
+                path = Path(original_path)  #: convert to a Path if it isn't already
+                if path.is_file():
+                    message.attach(self._build_gzip_attachment(path))
```

Comment:
**thought**: This creates a bit of arrow code and while it looks correct, I prefer the following.
```suggestion
            for original_path in message_details.attachments:
                if original_path is None: #: not sure if this needs to handle empty strings
                    continue

                path = Path(original_path)
                
                if not path.is_file():
                    continue
                
                message.attach(self._build_gzip_attachment(path))
```

Reply:
I was thinking the same thing but couldn't figure out how to fix it. I'll take this and the empty string stuff and rework it 👍